src/gcp_scanner/scanner.py

```diff
@@ -299,105 +299,21 @@ def main():
-      '-o',
-      '--output-dir',
-      required=True,
-      dest='output',
-      default='scan_db',
-      help='Path to output directory')
-
-  parser.add_argument(
-      '-k',
-      '--sa-key-path',
-      default=None,
-      dest='key_path',
-      help='Path to directory with SA keys in json format')
-  parser.add_argument(
-      '-g',
-      '--gcloud-profile-path',
-      default=None,
-      dest='gcloud_profile_path',
-      help='Path to directory with gcloud profile. Specify -\
- to search for credentials in default gcloud config path'
-  )
-  parser.add_argument(
-      '-m',
-      '--use-metadata',
-      default=False,
-      dest='use_metadata',
-      action='store_true',
-      help='Extract credentials from GCE instance metadata')
-  parser.add_argument(
-      '-at',
-      '--access-token-files',
-      default=None,
-      dest='access_token_files',
-      help='A list of comma separated files with access token and OAuth scopes.\
-TTL limited. A token and scopes should be stored in JSON format.')
-  parser.add_argument(
-      '-rt',
-      '--refresh-token-files',
-      default=None,
-      dest='refresh_token_files',
-      help='A list of comma separated files with refresh_token, client_id,\
-token_uri and client_secret stored in JSON format.'
-  )
-  parser.add_argument(
-      '-s',
-      '--service-account',
-      default=None,
-      dest='key_name',
-      help='Name of individual SA to scan')
-  parser.add_argument(
-      '-p',
-      '--project',
-      default=None,
-      dest='target_project',
-      help='Name of individual project to scan')
-  parser.add_argument(
-      '-f',
-      '--force-projects',
-      default=None,
-      dest='force_projects',
-      help='Comma separated list of project names to include in the scan')
-  parser.add_argument(
-      '-c',
-      '--config',
-      default=None,
-      dest='config_path',
-      help='A path to config file with a set of specific resources to scan.')
-  parser.add_argument(
-      '-l',
-      '--logging',
-      default='WARNING',
-      dest='log_level',
-      choices=('INFO', 'WARNING', 'ERROR'),
-      help='Set logging level (INFO, WARNING, ERROR)')
-
-  args: argparse.Namespace = parser.parse_args()
-
-  if not args.key_path and not args.gcloud_profile_path \
-    and not args.use_metadata and not args.access_token_files\
-    and not args.refresh_token_files:
-    logging.error(
-        'Please select at least one option to begin scan\
- -k/--sa-key-path,-g/--gcloud-profile-path, -m, -rt, -at'
-    )
+  args = arguments.arg_parser()
 
   force_projects_list = list()
   if args.force_projects:
     force_projects_list = args.force_projects.split(',')
 
-  logging.basicConfig(level=getattr(logging, args.log_level.upper(), None),
-                      format='%(asctime)s - %(levelname)s - %(message)s',
-                      datefmt='%Y-%m-%d %H:%M:%S')
-
+  if args.log_file:
+    logging.basicConfig(filename='../logs.log',
```

Comment:
I'd suggest to let user specify logdir path instead of making assumption that parent dir is available for writing.

Reply:
I am working on it such that the user specifies the logging directory and if it doesn't exist I create it and put the logs in it.
-ld logsfile
this will create the logsfile directory and adds the logs to it